types/src/bytesrepr.rs

```diff
@@ -1317,6 +1087,58 @@ where
     }
 }
 
+/// Serializes dynamic bytes array.
+pub fn serialize_bytes(bytes: &[u8]) -> Result<Vec<u8>, Error> {
```

Comment:
I think we should avoid `serialize_` here since we're used to that in the context of the `serialize()` method.  This applies to all the new functions in the section.  It's more important for the `deserialize_`, since the `deserialize()` function does the check for left over bytes, which these new functions don't.

I'd prefer if this was renamed to `vec_u8_to_bytes()`.

I also think it's be safer to change the signature to `vec_u8_to_bytes(bytes: &Vec<u8>)` so we can't accidentally pass a slice of an array in here.  I see we're using this function inside `to_bytes` of `String`, but we can keep this function as private to this module (again heavily documented to try and avoid misuse) and have `vec_u8_to_bytes(bytes: &Vec<u8>)` just call it.